lib/trino-collect/src/main/java/io/trino/collect/cache/EvictableCache.java

```diff
@@ -276,33 +312,106 @@ public void putAll(Map<? extends K, ? extends V> m)
             @Override
             public void clear()
             {
-                delegate.clear();
+                dataCacheMap.clear();
+                tokens.clear();
             }
 
             @Override
             public Set<K> keySet()
             {
-                return delegate.keySet();
+                return tokens.keySet();
             }
 
             @Override
             public Collection<V> values()
             {
-                // values() should be a view, but also, it has a size and, iterating values shouldn't throw for incomplete futures.
-                throw new UnsupportedOperationException();
+                return dataCacheMap.values();
             }
 
             @Override
-            public Set<Entry<K, V>> entrySet()
+            public Set<Map.Entry<K, V>> entrySet()
             {
                 throw new UnsupportedOperationException();
```

Comment:
Why is this operation unsupported? Could you please add meaningful message in the exception?

Reply:
I don't know how to implement it.